api/install/install.go

```diff
@@ -92,7 +97,76 @@ func Handler(service service) http.Handler {
 			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
+		resp, err := service.Install(r.Context(), req)
+		if err != nil {
+			respondInstallError(w, "error while installing chart: %v", err)
+			return
+		}
+
+		if err := json.NewEncoder(w).Encode(&resp); err != nil {
+			respondInstallError(w, "error writing response: %v", err)
+			return
+		}
+	})
+}
 
+// RestHandler handles an install request
+// swagger:operation PUT /releases/{kube_context}/{namespace}/{release_name} release installOperation
+//
+// Install helm release at the specified location
+//
+// ---
+// produces:
+// - application/json
+// parameters:
+// - name: kube_context
+//   in: path
+//   required: true
+//   default: minikube
+//   type: string
+//   format: string
+// - name: namespace
+//   in: path
+//   required: true
+//   default: default
+//   type: string
+//   format: string
+// - name: release_name
+//   in: path
+//   required: true
+//   type: string
+//   format: string
+// - name: Body
+//   in: body
+//   required: true
+//   schema:
+//    "$ref": "#/definitions/installRequestBody"
+// schemes:
+// - http
+// responses:
+//   '200':
+//    "$ref": "#/responses/listResponse"
+//   '400':
+//    "$ref": "#/responses/listResponse"
+//   '404':
+//    "$ref": "#/responses/listResponse"
+//   '500':
+//    "$ref": "#/responses/listResponse"
+
+func RestHandler(service service) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+
+		var req Request
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			logger.Errorf("[Install] error decoding request: %v", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		values := mux.Vars(r)
+		req.Flags.KubeContext = values["kube_context"]
+		req.Flags.Namespace = values["namespace"]
```

Comment:
namespace will always be defined since the path is `/releases/{cluster}/{namespace}/{release-name}`. So the handler will not be invoked unless the namespace is defined, which looks idiomatic to me.

> if the namespace's not defined we should fallback to the `default` namespace (if helm pkg doesn't do this, we can do this over here)